wgsl/index.bs

```diff
@@ -4081,6 +4081,72 @@ TODO: user data inputs of fragment stage must be subset of user data outputs of
 
 ### Input-output matching rules TODO ### {#input-output-matching}
 
+# Language extensions # {#language-extensions}
+
+The [SHORTNAME] language is expected to evolve over time.
+
+An <dfn noexport>extension</dfn> is a named grouping for a coherent
+set of modifications to a particular version of the [SHORTNAME] specification, consisting of any combination of:
+* Addition of new concepts and behaviours via new syntax, including:
+    * declarations, statements, attributes, and built-in functions.
+* Removal of restrictions in the current specification.
+* Syntax for reducing the set of permissible behaviours.
+* A description of how the feature interacts with the existing specification, and optionally with other extensions.
+
+For example, extensions could be used to:
+* Add numeric scalar types, such as 16-bit integers.
+* Add syntax to constrain floating point rounding mode.
+* Add new kinds of statements.
+* Add new built-in functions.
+* Add constraints on how shader invocations execute.
+* Add new shader stages.
+
+An <dfn noexport>enable declaration</dfn> indicates that a particular named
+feature may be used in the source text that follows.
+The declaration must be at module-scope.
+
+The declaration uses an identifier to name the feature, but does not
+introduce the identifier for later use.
+That is, the declaration does not cause the identifier to be in scope:
+the declaration does not conflict with any earlier or later declaration introducing that name.
+
+<pre class='def'>
+enable_decl
+  : ENABLE IDENT SEMICOLON
+</pre>
+
+Note: The grammar rule includes the terminating semicolon token,
```

Comment:
It is non-obvious to me that the enable must precede the use of the feature. I think that should be stated outside of a note.

Reply:
I thought that was covered by the phrasing above:

> ... may be used in the source text that follows.

Maybe I should strengthen that to:

> An enable declaration indicates that a particular extension is in effect in the source text that follows the declaration itself.  That is, language functionality described by the extension may be used in any source text
after the `enable` declaration.